plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/CompletionOptions.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.scripting.ide_services.compiler
+
+import kotlin.script.experimental.api.ScriptCompilationConfigurationKeys
+import kotlin.script.experimental.util.PropertiesCollection
+
+interface ReplCompletionOptionsKeys
+
+open class ReplCompletionOptionsBuilder : PropertiesCollection.Builder(), ReplCompletionOptionsKeys {
+    companion object : ReplCompletionOptionsKeys
+}
+
+@Suppress("unused")
+fun ReplCompletionOptionsBuilder.filterOutShadowedDescriptors(value: Boolean) {
+    this[filterOutShadowedDescriptors] = value
+}
+
+fun ReplCompletionOptionsBuilder.nameFilter(value: (String, String) -> Boolean) {
+    this[nameFilter] = value
+}
+
+val ReplCompletionOptionsKeys.filterOutShadowedDescriptors by PropertiesCollection.key(true)
+val ReplCompletionOptionsKeys.nameFilter
+        by PropertiesCollection.key<(String, String) -> Boolean>({ name, namePart -> name.startsWith(namePart) })
+
+@Suppress("unused")
```

Comment:
Is it really unused?
It is ok to have unused things if you explain why in comments, but it seems that in this case it will be safer to write a test that uses it.

Reply:
No, it's because of unused receiver. Unfortunately, I can't make suppress name more specific.
The test exists, obviously